File: 5th Semester/INF318/Examen3/anidados.pl
mostrar(I,J):- write(I),write("x"),write(J),P is I*J, write("="),write(P),nl.
mostrar3(I,J,K):- write(I),write("x"),write(J),write("x"),write(K),P is I*J*K, write("="),write(P),nl.

%tabla 
tabla(N):- tabla(N,1).

tabla(N,I):- I>N,!.
tabla(N,I):- I1 is I+1,
			 cj(N,I,1),
			 tabla(N,I1).
			 
cj(N,_,J):- J>N,!.
cj(N,I,J):- J1 is J+1,
			mostrar(I,J),
			cj(N,I,J1).
			
% tabla Ascendiente Descendiente
tablaAscDesc(N):- tablaAscDesc(N,1).

tablaAscDesc(N,I):- I>N,!.
tablaAscDesc(N,I):- I1 is I+1,
			 cj2(N,I,N),
			 tablaAscDesc(N,I1).
			 
cj2(_,_,J):- J<1,!.
cj2(N,I,J):- J1 is J-1,
			 mostrar(I,J),
			cj2(N,I,J1).
% tabla Pares			
tablaPares(N):- tablaPares(N,2).

tablaPares(N,I):- I>N,!.
tablaPares(N,I):- I1 is I+2,
			 cj3(N,I,2),
			 tablaPares(N,I1).
			 
cj3(N,_,J):- J>N,!.
cj3(N,I,J):- J1 is J+2,
			mostrar(I,J),
			cj3(N,I,J1).
			
% tabla de 3 argumentos
tabla3(N):- tabla3(N,1).

tabla3(N,I):- I>N,!.
tabla3(N,I):- I1 is I+1,
			 cj4(N,I,1),
			 tabla3(N,I1).
			 
cj4(N,_,J):- J>N,!.
cj4(N,I,J):- J1 is J+1,
			ck4(N,I,J,1),
			cj4(N,I,J1).
			
ck4(N,_,_,K):- K>N,!.
ck4(N,I,J,K):- K1 is K+1,
			mostrar3(I,J,K),
			 ck4(N,I,J,K1).

% tabla sin rep

						  
tablaSinRep(N):- tablaSinRep(N,1).

tablaSinRep(N,I):- I>N,!.
tablaSinRep(N,I):- I1 is I+1,
			 cj5(N,I,I),
			 tablaSinRep(N,I1).
			 
cj5(N,_,J):- J>N,!.
cj5(N,I,J):- J1 is J+1,
			ck5(N,I,J,J),
			cj5(N,I,J1).
			
ck5(N,_,_,K):- K>N,!.
ck5(N,I,J,K):- K1 is K+1,
			mostrar3(I,J,K),
			 ck5(N,I,J,K1).

File: 5th Semester/INF318/Examen3/anidados.py
# tabla

def mostrar(i, j):
	print(i, 'x', j, '\t=', i*j, '\n')
	
def mostrar3(i, j,k):
	print(i, 'x', j, 'x',k,'\t=', i*j*k, '\n')

# Tabla de Multiplicar ixj
def tabla(n):
	for i in range(1,n+1):
		for j in range(1,n+1):
			mostrar(i,j)

def tablarec(n, i):
	if (i <= n):
		cj(n, i, 1)
		tablarec(n, i+1)

			
def cj(n, i, j):
	if (j<=n):
		mostrar(i, j)
		cj(n, i, j+1)
			
# tabla Ascendente y Descendente ixj
def tablaAscDesc(n):
	for i in range(1, n+1, 1):
		for j in range(n, 0, -1):
			mostrar(i,j)
			
def tablaAscDescrec(n, i):
	if (i <= n):
		cj2(i, n)
		tablaAscDescrec(n, i+1)
		
def cj2(i, j):
	if (j > 0):
		mostrar(i, j)
		cj2(i, j-1)
		
# tabla pares ixj			
def tablaPares(n):
	for i in range(2, n+1,2):
		for j in range(2, n+1,2):
			mostrar(i, j)
			
			
def tablaParesrec(n, i):
	if (i <= n):
		cj3(n, i, 1)
		tablaParesrec(n, i+1)
		
def cj3(n, i, j):
	if (j <= n):
		if (j%2 == 0):
			mostrar(i, j)
		cj3(n, i, j+1)
# Tabla factores iguales ixjxk
def tablaIguales(n):
	for i in range(1, n+1):
		for j in range(1, n+1):
			for k in range(1, n+1):
				if (i==j and j==k):
					mostrar3(i, j, k)
					
def igual(a,b,c):
	if(a==b and b==c and c==a):
		return True
	else:
		return False

def tablaIgualesrec(n, i):
	if (i <= n):
		cj5(n, i, 1)
		tablaIgualesrec(n, i+1)

def cj5(n, i, j):
	if (j <= n):
		ck5(n, i, j, 1)
		cj5(n, i, j+1)

def ck5(n, i, j, k):
	if (k <= n):
		if (igual(i, j, k)):
			mostrar3(i, j, k)
		ck5(n, i, j, k+1)
					
# tabla sin repeticion de permutacion ixjxk
def tablaSinRepeticion(n):
        for i in range(1,n+1):
                for j in range(i,n+1):
                        for k in range(j,n+1):
                                mostrar3(i,j,k)
